fedimint-api-client/src/api.rs

```diff
@@ -1117,29 +1118,35 @@ pub struct WsFederationApi<C = WsClient> {
     module_id: Option<ModuleInstanceId>,
 }
 
-/// Some data shared/preserved between [`FederationPeerClient`] and
+/// Connection state shared/preserved between [`FederationPeerClient`] and the
 /// Jit tasks it spawns.
 #[derive(Debug)]
-struct FederationPeerClientShared {
-    last_connection_attempt: SystemTime,
-    connection_attempts: u64,
+struct FederationPeerClientConnectionState {
+    /// Last time a connection attempt was made, or `None` if no attempt has
+    /// been made yet.
+    last_connection_attempt_or: Option<SystemTime>,
```

Comment:
Any benefit of not initializing with current time or something just to avoid handling `None` in other places?

Reply:
I guess better logging is kind of nice.

Reply:
The `None` state symbolizes that we haven't yet attempted to connect. This code below is the only reason we need that state:

```
                if cs.last_connection_attempt_or.is_none() {
                    info!(
                    target: LOG_CLIENT_NET_API,
                    peer_id = %peer_id,
                    url = %url,
                    "Connecting to peer...");
                } else {
                    debug!(
                    target: LOG_CLIENT_NET_API,
                    peer_id = %peer_id,
                    url = %url,
                    "Retrying connecting to peer...");
                }
```

We could keep `last_connection_attempt` as a non-`Option<T>`, but then we'd need a boolean value to keep track of whether we've attempted a connection already. I think the way it is now is more intuitive but I can change it if you disagree.